gluten-core/src/main/scala/io/glutenproject/extension/StrategyOverrides.scala

```diff
@@ -69,8 +69,8 @@ object JoinSelectionOverrides extends Strategy with JoinSelectionHelper with SQL
     } else {
       // non equal condition
       // Generate BHJ here, avoid to do match in `JoinSelection` again.
-      val buildSide = getBroadcastBuildSide(left, right, joinType, hint, true, conf)
-        .orElse(getBroadcastBuildSide(left, right, joinType, hint, false, conf))
+      val isHintEmpty = hint.leftHint.isEmpty && hint.rightHint.isEmpty
```

Comment:
would it be more reasonable to use
```suggestion
      val isHintEmpty = !(hintToBroadcastLeft || hintToBroadcastRight)
```

Reply:
Oh I saw your comment under the topic
```
Fix SHUFFLE_HASH hint can convert BHJ gluten plan to SHJ.
```

By going through vanilla Spark's code:
```
        createBroadcastHashJoin(true)
          .orElse { if (hintToSortMergeJoin(hint)) createSortMergeJoin() else None }
          .orElse(createShuffleHashJoin(true))
          .orElse { if (hintToShuffleReplicateNL(hint)) createCartesianProduct() else None }
          .getOrElse(createJoinWithoutHint())
```
For this code snippet it seems to me that SHJ is created from logical Join but not BHJ, and is created in lower precedence comparing to BHJ.

Would you like to explain a bit more about how and why this patch implements "convert BHJ gluten plan to SHJ" ?

Reply:
isHintEmpty refer from Spark 3.3.1 JoinHint.isEmpty, Spark 3.2 does not implement the isEmpty method, so I copied the code. What I described may not be very clear. What I want to express is to fix the problem that the hint can take effect when adding the SHUFFLE_HASH hint, instead of converting BHJ to SHJ.

When I was testing, I wanted to convert a query to SHJ through hint, but found that it could not work in gluten, and found that there was a problem with this code.

Reply:
> would it be more reasonable to use

Do you think it should be modified to be more reasonable or consistent with the vanilla Spark code?

Reply:
I understand your case now. Thanks. let's utilize this patch now then we can continue optimizing to align with Spark.

Reply:
Hello @liujiayi771 , looks this change disables DPP in q1 query of TPCDS. Basically, all joins should be BHJ are converted to SMJ or SHJ. 'JoinHint' in this case is (Some(None), None), but your condition returns false :(.
cc @zhztheplayer.

Reply:
@zhztheplayer can you take a look , does our nightly either disabled?

Reply:
Seems like it (`hint.leftHint.isEmpty && hint.rightHint.isEmpty`) was Vanilla Spark's practice to decide whether a hint is empty or not:

https://github.com/apache/spark/blob/77b164aac9764049a4820064421ef82ec0bc14fb/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala#L65

@zhli1142015 Can the issue be reproduced in Vanilla Spark? We can compare the execution flow between Gluten Spark and Vanilla Spark to locate the root cause.

Reply:
@zhztheplayer , thank you for taking a look. I tested with OSS spark 3.2 and can't reproduce this issue. Seems '(Some(None), None)' JoinHint is introduced by our some internal changes, I will check with my team